main/src/test/java/org/mobilitydata/gtfsvalidator/validator/GtfsTripServiceIdForeignKeyValidatorTest.java

```diff
@@ -0,0 +1,146 @@
+import com.google.common.collect.ImmutableSet;
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Set;
+import org.junit.Test;
+import org.mobilitydata.gtfsvalidator.notice.ForeignKeyError;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendar;
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendarDate;
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendarDateTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendarTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsTrip;
+import org.mobilitydata.gtfsvalidator.table.GtfsTripTableContainer;
+import org.mobilitydata.gtfsvalidator.type.GtfsDate;
+import org.mobilitydata.gtfsvalidator.util.CalendarUtilTest;
+
+public class GtfsTripServiceIdForeignKeyValidatorTest {
+  static final Set<DayOfWeek> weekDays =
+      ImmutableSet.of(
+          DayOfWeek.MONDAY,
+          DayOfWeek.TUESDAY,
+          DayOfWeek.WEDNESDAY,
+          DayOfWeek.THURSDAY,
+          DayOfWeek.FRIDAY);
+
+  private static GtfsTripTableContainer createTripTable(
+      NoticeContainer noticeContainer, List<GtfsTrip> entities) {
+    return GtfsTripTableContainer.forEntities(entities, noticeContainer);
+  }
+
+  public static GtfsTrip createTrip(
+      long csvRowNumber, String routeId, String serviceId, String tripId, String shapeId) {
+    return new GtfsTrip.Builder()
+        .setCsvRowNumber(csvRowNumber)
+        .setRouteId(routeId)
+        .setServiceId(serviceId)
+        .setTripId(tripId)
+        .setShapeId(shapeId)
+        .build();
+  }
+
+  private static GtfsCalendarTableContainer createCalendarTable(
+      NoticeContainer noticeContainer, List<GtfsCalendar> entities) {
+
+    return GtfsCalendarTableContainer.forEntities(entities, noticeContainer);
+  }
+
+  public static GtfsCalendarDate createCalendarDate(
+      long csvRowNumber, String serviceId, GtfsDate date, int exceptionType) {
+    return new GtfsCalendarDate.Builder()
+        .setCsvRowNumber(csvRowNumber)
+        .setServiceId(serviceId)
+        .setDate(date)
+        .setExceptionType(exceptionType)
+        .build();
+  }
+
+  private static GtfsCalendarDateTableContainer createCalendarDateTable(
+      NoticeContainer noticeContainer, List<GtfsCalendarDate> entities) {
+
+    return GtfsCalendarDateTableContainer.forEntities(entities, noticeContainer);
+  }
+
+  @Test
+  public void tripServiceIdInCalendarTableShouldNotGenerateNotice() {
+    NoticeContainer noticeContainer = new NoticeContainer();
+    GtfsTripServiceIdForeignKeyValidator underTest = new GtfsTripServiceIdForeignKeyValidator();
+    underTest.calendarContainer =
+        createCalendarTable(
+            noticeContainer,
+            ImmutableList.of(
+                CalendarUtilTest.createGtfsCalendar(
+                    "WEEK", LocalDate.of(2021, 1, 14), LocalDate.of(2021, 1, 10), weekDays)));
```

Comment:
Looks like it's not a problem for your test to pass, but your start date comes after your end date. Maybe it would be better to invert it for coherence.

Reply:
Thanks! Good catch, FYI on real date this would have be caught by `CalendarServiceDateValidator`.